pkgs/os-specific/darwin/apple-sdk-11.0/default.nix

```
@@ -249,17 +249,24 @@ lib.genAttrs [
     "simd"
   ] mkStub;
 
-  inherit (pkgs)
-    callPackage
-    stdenv
-    llvmPackages_12
-    llvmPackages_13
-    llvmPackages_14
-    llvmPackages_15
-    llvmPackages_16
-    rustPlatform
-    xcodebuild
-    ;
-
   version = "11.0";
 }
+//
+  lib.genAttrs
+    [
+      "callPackage"
+      "stdenv"
+      "llvmPackages_12"
+      "llvmPackages_13"
+      "llvmPackages_14"
+      "llvmPackages_15"
+      "llvmPackages_16"
+      "rustPlatform"
+      "xcodebuild"
+    ]
+    (
+      name:
+      lib.warn
+        "apple_sdk_11_0.${name}: deprecated and will be removed in Nixpkgs 25.11; see <https://nixos.org/manual/nixpkgs/stable/#sec-darwin> for documentation and migration instructions"
```

```suggestion
        "darwin.apple_sdk_11_0.${name}: deprecated and will be removed in Nixpkgs 25.11; see <https://nixos.org/manual/nixpkgs/stable/#sec-darwin> for documentation and migration instructions"
```

Can you help me understand why these don’t cause warnings when `nix search` traverses into `llvmPackages_16` etc.?

---

That is a great question. Looking at `nix search --help`, I _think_ this is because this attrset lacks a `recurseForDerivations = true`, so `nix search` doesn't actually evaluate its values?

---

Right, makes sense.